src/plugins/intel_cpu/src/nodes/common/cpu_convert.cpp

```diff
@@ -55,6 +55,31 @@ void convert_vec<float, ov::float16>(jit_generator & gen,
     gen.movdqu(gen.xword[dst], f16vec);
 }
 
+// AVX512 instructions for ov::float16 to float conversion
+template <>
+void convert_vec<ov::float16, float>(jit_generator & gen,
+                                        const RegExp & src,
+                                        const RegExp & dst) {
+    auto const & f16vec = gen.zmm3;
+    auto const & f32vec = gen.zmm4;
+
+    gen.vmovdqu16(f16vec, gen.zword[src]);
+    gen.vcvtph2ps(f32vec, f16vec);
+    gen.vmovups(gen.zword[dst], f32vec);
+}
+
+template <>
+void convert_vec<float, ov::float16>(jit_generator & gen,
+                                        const RegExp & src,
+                                        const RegExp & dst) {
+    auto const & f16vec = gen.zmm3;
+    auto const & f32vec = gen.zmm4;
+
+    gen.vmovups(f32vec, gen.zword[src]);
+    gen.vcvtps2ph(f16vec, f32vec, 0);
+    gen.vmovdqu16(gen.zword[dst], f16vec);
+}
+
```

Comment:
When you do performance comparison between with AVX2 and AVX512, maybe you could change below lines to match vector size to adapt to avx512.

```
constexpr size_t vlen = 8u;
constexpr size_t vlen_log2 = 3;
```

Reply:
Do you have a unit test for accuracy and performance check please?

Reply:
> When you do performance comparison between with AVX2 and AVX512, maybe you could change below lines to match vector size to adapt to avx512.
> 
> ```
> constexpr size_t vlen = 8u;
> constexpr size_t vlen_log2 = 3;
> ```

Could you please try this to see if any benefits?